File: KangenApp/KangenApp/app/src/main/java/com/water/alkaline/kengen/data/db/repo/DownloadRepo.java
package com.water.alkaline.kengen.data.db.repo;

import android.app.Application;

import com.water.alkaline.kengen.data.db.AppDB;
import com.water.alkaline.kengen.data.db.dao.DownloadDao;
import com.water.alkaline.kengen.model.DownloadEntity;

import java.util.List;

public class DownloadRepo {
    DownloadDao dao;

    public DownloadRepo(Application application) {
        AppDB database = AppDB.getInstance(application);
        dao = database.downloadDao();
    }

    public void insert(DownloadEntity entity) {
        dao.insert(entity);
    }

    public void deleteAll() {
        dao.deleteAll();
    }

    public List<DownloadEntity> getAll() {
        return dao.getAll();
    }

    public List<DownloadEntity> getAllbyID(int data) {
        return dao.getAllbyID(data);
    }

    public List<DownloadEntity> getbyUrl(String data) {
        return dao.getbyUrl(data);
    }

    public List<DownloadEntity> getbyPath(String data) {
        return dao.getbyPath(data);
    }
}

File: KengenWaterApp/KengenWaterApp/app/src/main/java/com/water/alkaline/kengen/data/db/repo/DownloadRepo.kt
package com.water.alkaline.kengen.data.db.repo

import com.water.alkaline.kengen.data.db.dao.DownloadDao
import com.water.alkaline.kengen.model.DownloadEntity
import javax.inject.Inject

class DownloadRepo @Inject constructor(val dao: DownloadDao) {

    fun insert(entity: DownloadEntity) {
        dao.insert(entity)
    }

    fun deleteAll() {
        dao.deleteAll()
    }

    val all: MutableList<DownloadEntity>
        get() = dao.all

    fun getAllByID(data: Int): MutableList<DownloadEntity> {
        return dao.getAllbyID(data)
    }

    fun getByUrl(data: String): MutableList<DownloadEntity> {
        return dao.getbyUrl(data)
    }

    fun getByPath(data: String): MutableList<DownloadEntity> {
        return dao.getbyPath(data)
    }
}
